### DISCUSSION TITLE
Buzz: small language JIT compiled with MIR

### DISCUSSION THREAD

**Discussion by giann** [vnmakarov/mir#340] [Show and tell]
Hi there,

Just wanted to show you https://github.com/buzz-language/buzz
It's a small programming language written in Zig I'm working on.

I just finished implementing its JIT compiler with MIR!

**Comment by dibyendumajumdar** [comment]
Very nice

**Comment by vnmakarov** [comment]
Thank you for sharing this.  I believe MIR is a good solution for implementing static type language JITs.

Dynamic type languages needs a lot of additional work to implement performant MIR-based JIT (but it is true for any general purpose JIT compilers I know).
